### ISSUE TITLE
Switch the project extension.

### ISSUE THREAD

**Issue by Majoramari** [Redot-Engine/redot-engine#694]
### Tested versions

[83d6573](https://github.com/Redot-Engine/redot-engine/commit/83d65738b06986e5aa0943cae01e357133e04a75)

### System information

Non OS related

### Issue description

Switch the project extension from .godot to .redot

We could support both, or we could leave this issue open until we have a breaking separate us from the Godot engine so we apply it, or we just save a new project in .redot and add migration guide and one step in it to change the extension to .redot

### Steps to reproduce

None related

### Minimal reproduction project (MRP)

None related

**Comment by SkogiB** [comment]
Solid point. This was discussed some in the dev chat as well, with pretty much the same point you've iterated on a future break scenario. We'll have to get some organized discussion on this one.

**Comment by stalker57241** [comment]
What about special rdscript? It'll use .rd extension, working parallelly to gdscript, add requested in past method annotations, maybe `virtual` keyword to create virtual functions, some properties for `Callable`, like `.is_implemented()` for virtual functions, and so on...

**Comment by heppocogne** [comment]
How about renaming .godot/ folder to .redot/ ?
Renaming this folder might not cause user migration or compativity problem.

**Comment by SkogiB** [comment]
> What about special rdscript? It'll use .rd extension, working parallelly to gdscript, add requested in past method annotations, maybe `virtual` keyword to create virtual functions, some properties for `Callable`, like `.is_implemented()` for virtual functions, and so on...

We've discussed this, but it can't happen yet just from a compatibility point of view. It's definitely on the community's radar though

**Comment by CaCTuCaTu4ECKuu** [comment]
I don't think it could be a problem while there is easy and obvious way to convert between projects. While redot is fully compatible it's basically comes down to "Save as" that just duplicates cfg under another extension

**Comment by SkogiB** [comment]
Yeah but its just an extra little fuss we don't need to add for users coming over until we actually need to do it. Same stuff with changing extensions on gdscript and stuff.

**Comment by tindrew** [comment]
Closing this issue.
